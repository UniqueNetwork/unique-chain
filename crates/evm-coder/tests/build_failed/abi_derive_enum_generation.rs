use evm_coder_procedural::AbiCoder;

#[derive(AbiCoder)]
enum NonRepr {
	A,
	B,
	C,
}

#[derive(AbiCoder)]
#[repr(u32)]
enum NonReprU8 {
	A,
	B,
	C,
}

#[derive(AbiCoder)]
#[repr(u8)]
enum RustEnum {
	A(u128),
	B,
	C,
}

#[derive(AbiCoder)]
#[repr(u8)]
enum WithExplicit {
	A = 128,
	B,
	C,
}

fn main() {
	assert!(false);
}
