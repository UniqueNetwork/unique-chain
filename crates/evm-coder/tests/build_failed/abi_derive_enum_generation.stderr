error: Enum is not "repr(u8)"
 --> tests/build_failed/abi_derive_enum_generation.rs:4:6
  |
4 | enum NonRepr {
  |      ^^^^^^^

error: Enum is not "repr(u8)"
  --> tests/build_failed/abi_derive_enum_generation.rs:12:6
   |
12 | enum NonReprU8 {
   |      ^^^^^^^^^

error: Enumeration parameters should not have fields
  --> tests/build_failed/abi_derive_enum_generation.rs:21:2
   |
21 |     A(u128),
   |     ^

error: Enumeration options should not have an explicit specified value
  --> tests/build_failed/abi_derive_enum_generation.rs:29:2
   |
29 |     A = 128,
   |     ^
